includes/polling/ports/timos.inc.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+* timos.inc.php
+*
+* LibreNMS include timos (nokia) virtual router ports
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @package    LibreNMS
+* @link       http://librenms.org
+* @copyright  2018 Vitali Kari
+* @author     Vitali Kari <vitali.kari@gmail.com>
+*/
+
+// get all virtual router ports and statistics
+$timos_vrf_stats = snmpwalk_cache_twopart_oid($device, 'vRtrIfName', array(), 'TIMETRA-VRTR-MIB');
```

Comment:
This seems like a lot of snmpwalks.  It could be pretty slow, is there a table you can walk that will get all of this data?

Reply:
not really, the both Tables vRtrIfTable and vRtrIfStatsTable would take 2 seconds longer. (over a VPN connection)